lib/services/local/saml_idp_service_provider.go

```diff
@@ -164,6 +165,9 @@ func (s *SAMLIdPServiceProviderService) ensureEntityIDIsUnique(ctx context.Conte
 func validateSAMLIdPServiceProvider(sp types.SAMLIdPServiceProvider) error {
 	ed, err := samlsp.ParseMetadata([]byte(sp.GetEntityDescriptor()))
 	if err != nil {
+		if strings.Contains(err.Error(), "invalid url") {
+			return trace.BadParameter(err.Error())
```

Comment:
It's a shame we can't wrap err. If only there existed a superior semantic errors library...  :P